helix-core/src/main/java/org/apache/helix/monitoring/metrics/model/RatioMetric.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.helix.monitoring.metrics.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.SimpleDynamicMetric;
+
+
+/**
+ * A gauge which defines the ratio of one value to another.
+ */
+public abstract class RatioMetric extends SimpleDynamicMetric<Double> implements Metric<Double> {
```

Comment:
I think we just name it GaugeMetric with a <T> (it should be a template). There is no special operation been done regarding ratio.

Reply:
@jiajunwang 
I think creating `RatioMetric` is acceptable. Doesn't `GaugeMetric` explicitly extend `SimpleDynamicMetric<Long>`? Long != Double?

Reply:
As @pkuwm said, I think defining a wrapper class Ratio might be an overkill, but we'd still need either a float or a double to express the ratio, correct? @jiajunwang

Reply:
No strong preference here. Call it DoubleGuageMetric is awkward... But what you are going to do if you have another metric that is not Ratio but also in Double? You will see a lot of duplicate code...